src/plots/cartesian/axis_defaults.js

```diff
@@ -135,11 +135,22 @@ module.exports = function handleAxisDefaults(containerIn, containerOut, coerce,
         if(!containerOut.rangebreaks.length) {
             delete containerOut.rangebreaks;
         } else {
+            var n = 0;
             for(var k = 0; k < containerOut.rangebreaks.length; k++) {
-                if(containerOut.rangebreaks[k].pattern === DAY_OF_WEEK) {
+                var brk = containerOut.rangebreaks[k];
+                if(brk.pattern === DAY_OF_WEEK) {
                     containerOut._hasDayOfWeekBreaks = true;
-                    break;
+                    n++;
                 }
+
+                if(brk.pattern === HOUR) {
+                    containerOut._hasHourBreaks = true;
+                    containerOut._dayHours = Math.round((brk.bounds[1] - brk.bounds[0] + 24) % 24);
+                    n++;
+                }
+
+                // break when found all types
+                if(n === 2) break;
```

Comment:
what if the user includes two breaks of one type, THEN a break of the other type? I suppose I can theoretically imagine wanting two disjoint hour periods - a morning shift and an evening shift, with a midday break? I wouldn't worry about doing something "pretty" in this case, perhaps just set `_dayHours` to something like 1 or 2 so we go straight from day to hour ticks or perhaps just use the first break. But it definitely shouldn't stop us from noticing `_hasDayOfWeekBreaks`

And vice versa I guess... maybe someone wants to skip Sundays and Wednesdays? Again, weird, but shouldn't stop us from catching `_hasHourBreaks`